impl/spanner/testutil/testutil.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package testutil helps with tests against Spanner.
+package testutil
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	"cloud.google.com/go/spanner/spannertest"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	databasepb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+var testDBFlag = flag.String("test_db", "", "Fully-qualified database name to test against; empty means use an in-memory fake.")
+
+var dbCount struct {
+	sync.Mutex
+	count int
+}
+
+func dbName() string {
```

Comment:
nit: This function returns a constant name when the flag is set, but always a unique one when it's unset. The constantness of the first branch may cause an accidental assumption that the whole function is idempotent.

Since this function is used only in one place, how about:
1. Renaming it to `uniqueDbName`/`fakeDBName` or something indicating it is only for the fake case and/or it is not idempotent.
2. Don't read `testDBFlag` in it, but make it all the way down in `Client` function, since it's ultimately the one making the decision about the name to use. (see the code suggestion below)